source/server/config_validation/dispatcher.cc

```diff
@@ -18,7 +18,7 @@ Network::ClientConnectionPtr ValidationDispatcher::createClientConnection(
 
 Network::DnsResolverSharedPtr ValidationDispatcher::createDnsResolver(
     const std::vector<Network::Address::InstanceConstSharedPtr>&) {
-  NOT_IMPLEMENTED;
+  return Network::DnsResolverSharedPtr(dns_resolver_);
```

Comment:
nit: Can't you just return dns_resolver_ here? (You mighty need to change the pointer to be the base class type)

Reply:
Since the method name is "create", I wanted to bump up use_count for shared_ptr object. The caller may assume ownership of "created" object and dispose it. I wanted to avoid accidental deletion of dns_resolver_ when use_count drops to zero. I will see if I can simplify that.

Reply:
The use count will be bumped when you return the copy either way. Wrapping it doesn't do anything...

Reply:
Yes, you are right. Returning by value does ++, RVO does not intervene here. Thanks.